exercises/binary-search-tree/example.py

```diff
@@ -0,0 +1,69 @@
+class TreeNode(object):
+    def __init__(self, value):
+        self.value = value
+        self.left_node = None
+        self.right_node = None
+
+    def __str__(self):
+        return str(self.value)
+
+
+class BinarySearchTree(object):
+    def __init__(self):
+        self.root = None
+
+    def add(self, value):
+        if(self.root is None):
+            self.root = TreeNode(value)
+            inserted = True
+            return inserted
+        else:
+            inserted = False
+            cur_node = self.root
+
+            while not inserted:
+                if(value < cur_node.value):
+                    if(cur_node.left_node):
+                        cur_node = cur_node.left_node
+                    else:
+                        cur_node.left_node = TreeNode(value)
+                        inserted = True
+                        return inserted
+                elif(value > cur_node.value):
+                    if(cur_node.right_node):
+                        cur_node = cur_node.right_node
+                    else:
+                        cur_node.right_node = TreeNode(value)
+                        inserted = True
+                        return inserted
```

Comment:
Looking at this again, I just noticed that this method is returning `True`/`False` to indicate success/failure. Under what condition is failure expected to occur? From what I can see, this code will never return `False`, and would actually just enter an infinite loop if `value == cur_node.value`.

I think it would be worth adding a `BinarySearchTree.list` method to this exercise that returns a sorted list of values in the tree. A `collections.deque` would be most efficient for this, but I don't think that the tests should enforce its use, so the tests should probably convert the actual output to a list (eg. `assertEqual(list(self.bst.list()), [1, 2, 3])`).

Doing this would allow us to check that values are placed in the right order - each test for adding a number should contain a second assert for the current list of stored values. We should also test duplicate values, because the README specifies a way for these to be handled.

**From the `README`:**
> All data in the left subtree is less than or equal to the current node's data, and all data in the right subtree is greater than the current node's data.

Reply:
@N-Parsons Returning `True` from an add or insert method is a common practice for container classes. I agree that it doesn't make the most sense, but that is likely why @rodrigocam did this.

Reply:
@cmccandless is right, I thought this could be good to give more safety. I made the list method that @N-Parsons asked and change the tests to use this new method. I fix the infinite loop condition that you point but I keep the bool return, if it was a problem I can remove (:

Reply:
Given that there's no way that `False` can be returned, and there is not test for this method returning `False`, I think it's unnecessary and likely to cause some confusion/frustration. I think it's fine for this example code to return `True`/`False`, but I don't think that it should be tested for.

Reply:
Done! I take out the return of the method.